products/libllbuild/BuildDB-C-API.cpp

```diff
@@ -0,0 +1,288 @@
+//
+//  BuildDB-C-API.c
+//  libllbuild
+//
+//  Created by Benjamin Herzog on 4/5/19.
+//  Copyright © 2019 Apple Inc. All rights reserved.
+//
+
+#include <llbuild/llbuild.h>
+
+#include "llbuild/Core/BuildDB.h"
+
+#include "llvm/Support/raw_ostream.h"
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+
+using namespace llbuild;
+using namespace llbuild::core;
+
+namespace {
+
+class CAPIBuildDBResultKeys {
```

Comment:
Supposed to be copyable? If not:

    CAPIBuildDBResultKeys(const CAPIBuildDBResultKeys&) LLBUILD_DELETED_FUNCTION;
    CAPIBuildDBResultKeys& operator=(const CAPIBuildDBResultKeys&) LLBUILD_DELETED_FUNCTION;

Repeats for other newly-introduced types.

(and this should be generalized into a reusable macro too)

Reply:
FYI @jakepetroules -- I'd recommend holding off on the small technical details like this until the overall design picture is reviewed. Your comments are all accurate, but its more important to get the big picture design right first; and sometimes the small comments can distract from that.